feat(dcgw): Network architecture (#4827)
Kong/developer.konghq.com: app/dedicated-cloud-gateways/network-architecture.md

---
title: "Dedicated Cloud Gateways network architecture"
content_type: reference
layout: reference
description: "Learn the architecture of Kong-managed Dedicated Cloud Gateways networks, how they communicate with your cloud infrastructure, and what you must decide before deploying a Dedicated Cloud Gateway."

products:
    - gateway
breadcrumbs:
  - /dedicated-cloud-gateways/
works_on:
  - konnect

related_resources:
  - text: Dedicated Cloud Gateways 
    url: /dedicated-cloud-gateways/
  - text: Dedicated Cloud Gateways private network architecture and security
    url: /dedicated-cloud-gateways/private-network/
  - text: Dedicated Cloud Gateways public network architecture and security
    url: /dedicated-cloud-gateways/public-network/
  - text: Multi-cloud Dedicated Cloud Gateway network architecture and security
    url: /dedicated-cloud-gateways/multi-cloud/
tags:
  - dedicated-cloud-gateways
---

> In a Dedicated Cloud Gateway, Kong manages the gateway infrastructure (compute, Dedicated Cloud Gateway network, and data planes) for you in a single-tenant cloud environment dedicated to your organization (AWS, GCP, Azure).

Reviewer: ```suggestion
In a Dedicated Cloud Gateway, Kong manages the Gateway infrastructure (compute, Dedicated Cloud Gateway network, and data planes) for you in a single-tenant cloud environment dedicated to your organization (AWS, GCP, Azure).
```

A complete Dedicated Cloud Gateway deployment consists of this Kong-managed network infrastructure and the cloud infrastructure with the upstream services you manage.

The following diagram shows how traffic flows in a Dedicated Cloud Gateway:
<!--vale off-->
{% mermaid %}
flowchart LR
    A(API consumers) --> |inbound edge| B(Kong data planes)
    B --> |upstream path| C(Your services)
{% endmermaid %}
<!--vale on-->

The networking decisions you make govern both hops independently.

Before you deploy a Dedicated Cloud Gateway, you must make some choices to determine how to deploy it based on your network and upstream service configuration:

> 1. Decide which cloud provider or providers (AWS, Azure, GCP) you want to use based on where your upstream service cloud infrastructure is currently deployed.

Reviewer: Do we need to list the cloud providers here? In the future providers could be added/removed so it's better to be generic unless this information is otherwise hard to find.
```suggestion
1. Decide which cloud provider or providers you want to use based on where your upstream service cloud infrastructure is currently deployed.
```

Reviewer: The table in 'Public and private network connectivity' should be enough.

1. Decide if you want a [public or private Dedicated Cloud Gateway](#public-and-private-network-connectivity) (or both) depending on if your upstream traffic is public or private.
1. For private Dedicated Cloud Gateways:
   * Decide how {{site.base_gateway}} will connect to your upstream services via private network peering (VPC/VNET), hub-and-spoke networking (Transit gateway, VWAN), or private endpoints (AWS resource endpoints).

>    * Decide how to resolve hostnames, either via private DNS or outbound DNS resolver (when your hostnames live on a separate DNS server).

Reviewer: ```suggestion
   * Decide how to resolve hostnames, either via private DNS or an outbound DNS resolver (when your hostnames live on a separate DNS server).
```

1. Decide if you need a [multi-cloud Dedicated Cloud Gateway](#multi-cloud-architecture) for high-availability.

## Kong-managed gateway infrastructure architecture

> The Kong-managed gateway infrastructure consists of data plane nodes that run inside of a Kong-managed network peering (VPC or VNET depending on your provider).

Reviewer: Should 'gateway' be capitalised here and elsewhere outside since it's the proper name of a product ?

> The Kong-managed gateway infrastructure consists of data plane nodes that run inside of a Kong-managed network peering (VPC or VNET depending on your provider).

Author: ```suggestion
The Kong-managed Gateway infrastructure consists of data plane nodes that run inside of a Kong-managed network peering (VPC or VNET depending on your provider).
```

The Kong-managed data plane nodes automatically scale with your throughput. 

The following diagram shows what the Kong-managed architecture looks like if you chose AWS as your provider:
<!--vale off-->
{% mermaid %}
flowchart LR
    subgraph kong_account["Kong-managed AWS infra"]
        subgraph kong_vpc["Kong-managed VPC"]
          dp1["<img src="/assets/icons/gateway.svg" style="max-height:20px"> Data plane node"]
          dp2["<img src="/assets/icons/gateway.svg" style="max-height:20px"> Data plane node"]
          dp3["<img src="/assets/icons/gateway.svg" style="max-height:20px"> Data plane node"]
        end
    end

    style kong_account stroke-dasharray:3,rx:10,ry:10
{% endmermaid %}
<!--vale on-->

> In a Dedicated Cloud Gateway, the Kong-managed infrastructure would communicate with your own cloud infrastructure that contains your databases, APIs, and other resources.

Reviewer: ```suggestion
In a Dedicated Cloud Gateway, the Kong-managed infrastructure communicates with your own cloud infrastructure that contains your databases, APIs, and other resources.
```

Load balancing happens at the connection level, not per request.
Multiple requests sent over the same TCP connection go to the same data plane node. 
Different connections distribute across nodes.

## Public and private network connectivity

Before you deploy a Dedicated Cloud Gateway, you'll need to determine if you should deploy a public or private gateway, or both.
Which you deploy depends on how API consumers are reaching your gateway and how you want data planes to reach your backend services:

* Use [public](/dedicated-cloud-gateways/public-network/) if:
  * You expose services to the internet with custom access control
  * Want minimal setup and are securing at the Kong layer
  * You're proxying your own infrastructure over the public internet
* Use [private](/dedicated-cloud-gateways/private-network/) if:
  * Your upstream services are on one or more private network (VPC or VNET)
  * You don't expose services to the public internet
  * Require network isolation or full edge control

>   * Operate under regulated security requirements

Reviewer: ```suggestion
* Use [public](/dedicated-cloud-gateways/public-network/) if you:
  * Expose services to the internet with custom access control
  * Want minimal setup and are securing at the Kong layer
  * Proxy your own infrastructure over the public internet
* Use [private](/dedicated-cloud-gateways/private-network/) if you:
  * Deploy upstream services on one or more private network (VPC or VNET)
  * Don't expose services to the public internet
  * Require network isolation or full edge control
  * Operate under regulated security requirements
```
avoid repetition of 'you' on every line

## Dedicated Cloud Gateway network CIDR range

Before you create a Dedicated Cloud Gateway network, determine which CIDR range you want to use for your network.
A CIDR block defines the range of IP addresses available for your Dedicated Cloud Gateway. 
If you're configuring private network connectivity, this CIDR block **must not** overlap with CIDR blocks assigned in your own cloud service provider networks to prevent conflicts.
Keep in mind that your Dedicated Cloud Gateway network CIDR block must be large enough to cover the Kong infrastructure Kong will provision inside it, such as the data plane nodes, the DNS proxy, internal load balancers, and any other components Kong manages. 

Keep the following CIDR requirements in mind when you're deciding your network CIDR range:
* **Prefix length:** The CIDR block must have a prefix length between `/16` and `/23`. `/23` blocks are only supported for up to 3 availability zones.
* **Private IP Range:** The entire CIDR block must fall within one of these private IP ranges:
  * 10.0.0.0/8
  * 100.64.0.0/10
  * 172.16.0.0/12
  * 192.168.0.0/16
  * 198.18.0.0/15
  
  {:.info}

>   > **Acceptable CIDR examples:**

Reviewer: Consider moving this callout to after the full list. Embedding it here looks awkward visually.

{% navtabs "dcgw-network" %}
{% navtab "UI" %}
1. In the {{site.konnect_short_name}} sidebar, click **Network**.
1. Click **New Network**.
1. From the **Provider** dropdown menu, select the cloud provider you want to deploy your Dedicated Cloud Gateway in.
1. From the **Region** dropdown menu, select the region you want to configure the cluster in. 
1. In the **Network name** field, enter a unique display name for your network.
1. In the **CIDR for Dedicated Cloud** field, enter your CIDR block.

   {:.danger}
   > **Important:** Your CIDR block **must not** overlap with any IP ranges already in use by your organization. Overlapping ranges can prevent network peering from functioning correctly. The default range is `10.0.0.0/16`, but this can be edited.
1. Click **Save**.
{% endnavtab %}
{% navtab "API" %}
1. Send a `GET` request to the [`/cloud-gateways/provider-accounts` endpoint](/api/konnect/cloud-gateways/v2/#/operations/list-provider-accounts) to get all cloud provider IDs:
{% konnect_api_request %}
url: /v2/cloud-gateways/provider-accounts
method: GET
region: global
{% endkonnect_api_request %}

> 1. In the response, copy and export the ID for the cloud provider you want to use for your Dedicated Cloud Gateway network:

Reviewer: Numbering is broken here:

<img width="961" height="685" alt="Image" src="https://github.com/user-attachments/assets/94e7ceae-8e2e-4afa-b742-f5f707a44cff" />
